kit/transport/http/error_handler.go

```diff
@@ -28,10 +35,11 @@ func (h ErrorHandler) HandleHTTPError(ctx context.Context, err error, w http.Res
 
 	code := errors2.ErrorCode(err)
 	var msg string
-	if err, ok := err.(*errors2.Error); ok {
+	if _, ok := err.(*errors2.Error); ok {
 		msg = err.Error()
 	} else {
-		msg = "An internal error has occurred"
+		msg = "An internal error has occurred - check server logs"
+		h.logger.Warn("internal error not returned to client", zap.Error(err))
```

Comment:
@dgnorton suggested (and I agree) it is probably too insecure to return any and every error back to the client. Instead, log the error and tell the client to check the server logs - anyone with server log access should have permission to see whatever errors we create.

Reply:
Particular errors that we actually want to return to the client should be converted to the correct format to be returned.

Reply:
New error from client:

```
Error: failed to backup bucket data: failed to download snapshot of shard 1: InfluxDB OSS-only command failed: 500 Internal Server Error: An internal error has occurred - check server logs
```

From server logs:

```
2021-09-13T15:45:36.828559Z     warn    internal error not returned to client   {"log_id": "0WZyCYXW000", "handler": "error_logger", "error": "error creating tsm hard link: hard coded failure for /Users/sarnold/.influxdbv2/engine/data/b029e667ccf6262d/autogen/1/000000015-000000002.tsm"}
```